### PR TITLE
Fix #22 Task 2.1: Store (new class)

### LINKED ISSUE #22: Task 2.1: Store (new class)
Napisz klasę Store, która będzie umożliwiała dokonywanie zakupów. Wykorzystaj poniższy
enum i funkcje.

- [x] enum class Response {done, lack_of_money, lack_of_cargo, lack_of_space};
- [ ] Response buy(Cargo* cargo, size_t amount, Player* player);
- [ ] Response sell(Cargo* cargo, size_t amount, Player* player);

### PR DESCRIPTION
**Pull request by madebykj**

### CODE REVIEW

File: shm/main.cpp [hunk 1]
```diff
@@ -37,19 +37,108 @@ void SampleTest1() {
...
+#include "Alcohol.hpp"
+#include "Cargo.hpp"
+#include "Fruit.hpp"
+#include "Item.hpp"
 #include "Store.hpp"
-void OperatorTest1() {
-    Store store;
-    store.CreateTestCargo();
-    std::cout << store << "\n";
+void SellBuyTest1() {
```
Comment (lksobczyk) [inline shm/main.cpp]: Maybe it should be in a separate file

**Review by lksobczyk — APPROVED** [review]
Magnificent

File: shm/Store.cpp [hunk 2]
```diff
@@ -2,30 +2,76 @@
...
 
-// Temporary added - operator << test
-//begin
 #include "Alcohol.hpp"
 #include "Fruit.hpp"
 #include "Item.hpp"
-//end
+
+Cargo* Store::FindCargo(Cargo* cargo) const {
+    auto match_cargo = std::find_if(std::begin(_cargo), std::end(_cargo),
```
Comment (Tomonike) [inline shm/Store.cpp]: camel Case

File: shm/Store.cpp [hunk 3]
```diff
@@ -2,30 +2,76 @@
...
-
+    const size_t price = amount * cargo->getPrice();
     if (price > player->getMoney()) {
+        std::cout << "player, lack of money\n";
         return Response::lack_of_money;
     }
-
+    if (auto* match_cargo = FindCargo(cargo)) {
+        if (amount > match_cargo->getAmount()) {
+            std::cout << "Store, lack of cargo\n";
```
Comment (Tomonike) [inline shm/Store.cpp]: camel Case

File: shm/Store.cpp [hunk 4]
```diff
@@ -2,30 +2,76 @@
...
     return Response::done;
 }
 
 Store::Response Store::Sell(Cargo* cargo, size_t amount, Player* player) {
+    const size_t price = amount * cargo->getPrice();
+    if (amount > cargo->getAmount()) {
+        std::cout << "cargo, lack of cargo\n";
+        return Response::lack_of_cargo;
+    }
+    if (auto* match_cargo = FindCargo(cargo)) {
```
Comment (Tomonike) [inline shm/Store.cpp]: camel Case

File: shm/Store.hpp [hunk 5]
```diff
@@ -17,11 +17,8 @@ class Store {
...
 
     friend std::ostream& operator<<(std::ostream& out, const Store& store);
-    // Temporary added - operator << test
-    //begin
-    void CreateTestCargo();
-    //end
 
 private:
     std::vector<std::shared_ptr<Cargo>> _cargo;
+    Cargo* FindCargo(Cargo* cargo) const;
```
Comment (Tomonike) [inline shm/Store.hpp]: camel Case

File: shm/main.cpp (hunk 1 above)
Comment (Tomonike) [inline shm/main.cpp]: camel Case

File: shm/main.cpp [hunk 6]
```diff
@@ -37,19 +37,108 @@ void SampleTest1() {
...
+    std::cout << "itm3 name: " << itm3.getName() << ", amount: " << itm3.getAmount() << ", base price: " << itm3.getBasePrice() << ", price: " << itm3.getPrice() << ", rarity: " << itm3.getTestRarity() << "\n";
+
+    std::cout << "\n";
 }
 //end
 
 int main() {
     SampleTest1();
-    OperatorTest1();  // Temporary added - operator << test
+    SellBuyTest1();  // Temporary added - sell / buy test
```
Comment (Tomonike) [inline shm/main.cpp]: camel Case

**Review by Tomonike — APPROVED** [review]
We stick to the camel case convention :)

**Pull request merged**